lib/Driver/ToolChains/Flang.cpp

```diff
@@ -38,6 +42,45 @@ using namespace clang::driver::tools;
 using namespace clang;
 using namespace llvm::opt;
 
+class FlangMacroBuilder : public MacroBuilder {
+  ArgStringList &CmdArgs;
+  const ArgList &DriverArgs;
+  public:
+    FlangMacroBuilder(ArgStringList &UpperCmdArgs, const ArgList &DriverArgs, llvm::raw_string_ostream &Output)
+      : MacroBuilder(Output), CmdArgs(UpperCmdArgs), DriverArgs(DriverArgs) {
+    }
+    virtual void defineMacro(const Twine &Name, const Twine &Value = "1") {
+      CmdArgs.push_back("-def");
+      CmdArgs.push_back(DriverArgs.MakeArgString(Name + Twine('=') + Value));
+    }
+    virtual void undefineMacro(const Twine &Name) {
+    }
+    virtual void append(const Twine &Str) {
+    }
+};
+
+// Following 2 functions are copied from lib/Frontend/InitPreprocessor.cpp
```

Comment:
Why not reuse the original?

Reply:
They are `static` functions in a `cpp` not meant as public functions. Shall I expose them as public functions?

Reply:
Yes, in case the original functions change in the future, we would not want to manually change the flang-driver copies to keep them in sync.

Reply:
Done